sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/AppConfigurationAutoConfiguration.java

```diff
@@ -21,14 +22,17 @@
 @ConditionalOnProperty(prefix = AppConfigurationProperties.CONFIG_PREFIX, name = "enabled", matchIfMissing = true)
 public class AppConfigurationAutoConfiguration {
 
+    /**
+     * Auto Watch
+     */
     @Configuration
     @ConditionalOnClass(RefreshEndpoint.class)
     static class AppConfigurationWatchAutoConfiguration {
 
         @Bean
         public AppConfigurationRefresh getConfigWatch(AppConfigurationProperties properties,
-                AppConfigurationProviderProperties appProperties, ClientStore clientStore) {
-            return new AppConfigurationRefresh(properties, appProperties, clientStore);
+            AppConfigurationProviderProperties appProperties, AppConfigurationReplicaClientFactory clientFactory) {
```

Comment:
These two properties have been passed to multiple places, can we narrow the scope for them?

Reply:
I have been trying to work on this, the issue is one is the customer provided properties, the other is our own properties files. Both of which are used in multiple locations.

Reply:
It probably won't be in this PR, but if we can define some inner classes in `AppConfigurationProperties`, one of them be `Refresh`. Then we can pass the `Refresh` object to construct this `AppConfigurationRefresh`.

Reply:
And can the `AppConfigurationReplicaClientFactory` provide this `configStores` information that the `AppConfigurationRefresh` needs? If so, the `AppConfigurationRefresh` would only require the `refresh` properties.

Reply:
We can start by defining the model these classes need when construction, like `AppConfigurationRefresh` and `KeyVaultClient`, then we can find a way to refactor instead of passing these two properties around.

Reply:
I did a bit of rework on this, I am not sure what you intended by the first part, but `ConnectionManager` has an instance of the config store in it, so I made that usable so `AppConfigurationProperties` could be removed.